### DISCUSSION TITLE
SRP service

### DISCUSSION THREAD

**Discussion by tyler-wynd** [openthread/openthread#10727] [Q&A] [answered]
I am setting up my devices to broadcast SRP service, each device has a unique SRP service. 
I am also using 2 border routers to scan the SRP to connect and receive data from the device. But when I turn on the 2nd Border router, the 1st Border router cannot scan any SRP service by command: `sudo ot-ctl srp server service`.
We are using Raspberry PI as Border router, is there any missing setup for Openthread on Linux?
the mDNS tool like avahi-browse, it can scan all SRP service but without IPv6 address so we cannot use.

**Comment by abtink** [comment]
@tyler-wynd, what you're observing is expected behavior. An SRP client registers with a single server. In this case, the Thread Border Router (TBR) acting  an SRP server also acts an Advertising Proxy, advertising services via mDNS on the infrastructure network. This allows service discovery to be performed through mDNS regardless of which SRP server the client registers with.

For your information, there are ongoing efforts to implement "SRP replication," which will enable SRP servers to synchronize registered information. This is primarily intended as a backup mechanism and is not currently available in `ot-br-posix`. Even with SRP replication, service discovery should always be performed via mDNS on the infrastructure network, rather than by directly querying SRP servers.

**Comment by tyler-wynd** [comment]
Hi @abtink ,
The border router is the client not server, the SRP server is my device. There 2 border routers with the same configuration in our network. but only 1 border router can scan my device.

**Reply by abtink** [reply]
**@tyler-wynd**

* To reiterate, an SRP client, regardless of the device type (BR, end-device, ect) it's running on,  will register with a single SRP server.
* Therefore, if you enable the SRP server function on multiple devices within your mesh network, the client-registered service will only appear on one of those SRP servers.
    * As previously noted, the SRP server should also function as an advertising proxy, avertising the service on the infrastructure interface via mDNS.
* It is unclear on what you mean by "only 1 border router can scan my device." The term "scan" can have various meanings (discovery scan, MAC scan, etc.), but it seems you might be referring to "service discovery."
    * It's important to remember that `ot-ctl srp server service` is an SRP server CLI command that lists all registered services with the SRP server running on that specific device. It doesn't query or "scan" for services.

BTW, to keep our conversation organized, you can reply directly in the box below my previous answer instead of using "Suggest an answer," which creates a separate chat entry.

**Reply by tyler-wynd** [reply]
@abtink I see, do you have command suggestion to discover the SRP service instead of using "ot-ctl srp server service", with the same result?

**Reply by jwhui** [reply, marked as answer]
I think there's some confusion on terminology.
- SRP Client is responsible for publishing DNS-SD records.
- SRP Server is responsible for maintaining a database of DNS-SD records from SRP Clients.

Typically, the SRP Server function is located on border routers so that they can proxy and re-publish the DNS-SD records on Wi-Fi / Ethernet using mDNS.

As @abtink mentioned, an SRP Client will only publish its DNS-SD records to a single SRP Server.
